lib/ex_doc/retriever.ex

```diff
@@ -193,6 +193,7 @@ defmodule ExDoc.Retriever do
     specs = all_specs
             |> Map.get(function, [])
             |> Enum.map(&Typespec.spec_to_ast(name, &1))
+            |> Enum.sort( fn({_,l,_}, {_,l2,_}) -> l[:line] < l2[:line] end )
```

Comment:
I wonder if they are just in reverse order and calling `Enum.reverse` would be enough?